packages/@vuepress/core/lib/app/app.js

```diff
@@ -62,15 +62,25 @@ export function createApp (isServer) {
     base: siteData.base,
     mode: 'history',
     fallback: false,
-    routes
+    routes,
+    scrollBehavior (to, from, savedPosition) {
+      if (savedPosition) {
+        return savedPosition
+      }
+      if (to.path !== from.path) {
+        return { x: 0, y: 0 }
+      }
+    }
   })
 
   // redirect /foo to /foo/
   router.beforeEach((to, from, next) => {
     if (!/(\/|\.html)$/.test(to.path)) {
-      next(Object.assign({}, to, {
-        path: to.path + '/'
-      }))
+      next(
```

Comment:
Do not format code and only focus what you are doing.

Reply:
Ah sorry, I'm using prettier by default.